### PR TITLE
Adding NLVR model

### PR DESCRIPTION
**Pull request by matt-gardner**
This is taking the code as-is from the `wikitables` branch.  I didn't change anything, except for the location of the modules - I thought `models/semantic_parsing/` made more sense than `models/encoder_decoders/` (and it's more aligned with how we've started to organize other groups of models).

### CODE REVIEW

File: tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/denotations.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+false
```
Comment (pdasigi) [inline tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/denotations.txt]: Did you mean to commit an untar-ed `model.tar.gz`?

File: tests/fixtures/semantic_parsing/nlvr_direct_semantic_parser/serialization/vocabulary/denotations.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+true
```
Comment (pdasigi) [inline tests/fixtures/semantic_parsing/nlvr_direct_semantic_parser/serialization/vocabulary/denotations.txt]: Same here.

File: tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/denotations.txt (hunk 1 above)
Comment (matt-gardner) [reply tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/denotations.txt]: This is what gets generated when you run `train_fixtures.py`.  I don't think it's a problem to have this here; seeing the vocabulary files from the test fixtures can actually be helpful.

Comment (pdasigi) [reply tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/denotations.txt]: Good to know. I agree that this is useful to look at, just wasn't sure if this was intended to be checked it.

File: tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/non_padded_namespaces.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+actions
```
Comment (pdasigi) [inline tests/fixtures/semantic_parsing/nlvr_coverage_semantic_parser/serialization/vocabulary/non_padded_namespaces.txt]: You may want to regenerate the fixtures though. This seems outdated. The non-padded namespaces should be `denotations` and `rule_labels`. I fixed that in the config files recently.

**Pull request merged**